ledger/complete/checkpointqueue.go

```diff
@@ -0,0 +1,87 @@
+package complete
+
+import (
+	"github.com/onflow/flow-go/ledger/complete/mtrie/trie"
+)
+
+// CheckpointQueue is a fix-sized FIFO queue of MTrie.
+// CheckpointQueue is intentionally not threadsafe given its limited use case.
+// CheckpointQueue is not a general purpose queue to avoid incurring overhead
+// for features not needed for its limited use case.
+type CheckpointQueue struct {
```

Comment:
CheckpointQueue sounds like a queue with checkpointing files. 

If we want to name it with behavior, maybe we could call it `TrieQueue`. 

If we want to name it with functionality, maybe `LatestWrittenTries` would be more accurate. It holds the latest trie whose trie update have been written to the WAL file.

Reply:
Good idea!   Renamed `CheckpointQueue` to `TrieQueue` as suggested in https://github.com/onflow/flow-go/pull/2792/commits/853ff0545636870a21393258e81c986dccdbfc4a.  

Since the name TrieQueue is not specific to checkpointing, also moved it from `complete` package to `wal` package.

Putting it in the `wal` package makes it more clear `TrieQueue` is specific to checkpointing and can prevent it from being used for other purposes, getting modified, and introducing unwanted side-effects to checkpointing.